x/emissions/module/rewards_v2_test.go

```diff
@@ -0,0 +1,793 @@
+				{0.01635, 0.00179, 0.03396, 0.0153, 0.01988, 0.00962, 0.01191, 0.01616, 0.01417, 0.01216, 0.01292, 0.01564, 0.01323, 0.01261, 0.01145, 0.0163, 0.014, 0.01373, 0.01453, 0.01207, 0.01641, 0.01601, 0.01114, 0.01259, 0.01589, 0.01229, 0.01309, 0.0138, 0.01162, 0.01145, 0.01013, 0.01208, 0.0111, 0.0118, 0.01374, 0.01428, 0.01791, 0.01288, 0.01161, 0.01151, 0.01148, 0.01284, 0.01239, 0.01023, 0.01712, 0.0116, 0.01639, 0.01043, 0.01308, 0.01455, 0.01607, 0.01205, 0.01357, 0.01108, 0.01633, 0.01208, 0.01278},
+				{0.01345, 0.00209, 0.03249, 0.01688, 0.02126, 0.01338, 0.0116, 0.01605, 0.0133, 0.01407, 0.01367, 0.01244, 0.0145, 0.01262, 0.01348, 0.01684, 0.01148, 0.01705, 0.01714, 0.0124, 0.0125, 0.01462, 0.01274, 0.01407, 0.01667, 0.01316, 0.01628, 0.01373, 0.01409, 0.01603, 0.01378, 0.01143, 0.013, 0.01644, 0.01528, 0.01441, 0.01404, 0.01402, 0.01479, 0.01417, 0.01244, 0.0116, 0.01419, 0.01497, 0.01629, 0.01514, 0.01133, 0.01339, 0.01053, 0.01424, 0.01428, 0.01446, 0.01805, 0.01229, 0.01586, 0.01234, 0.01513},
+				{0.01675, 0.00318, 0.02623, 0.02734, 0.03526, 0.02733, 0.01697, 0.01619, 0.01925, 0.02018, 0.01735, 0.01922, 0.02225, 0.0189, 0.01923, 0.03193, 0.01956, 0.01763, 0.01975, 0.01466, 0.02021, 0.01803, 0.01438, 0.01929, 0.02305, 0.02223, 0.02445, 0.01967, 0.02292, 0.01878, 0.01751, 0.02695, 0.01849, 0.01658, 0.01948, 0.01594, 0.02318, 0.01906, 0.01607, 0.01369, 0.01686, 0.01314, 0.01936, 0.01518, 0.018, 0.02212, 0.02259, 0.01674, 0.02944, 0.01796, 0.02187, 0.01895, 0.01637, 0.01594, 0.01608, 0.02203, 0.01486},
+				{0.02093, 0.00213, 0.02462, 0.0203, 0.03115, 0.01, 0.01545, 0.01785, 0.01662, 0.01156, 0.02284, 0.01475, 0.01331, 0.01592, 0.01462, 0.02333, 0.01836, 0.01465, 0.0186, 0.01566, 0.01506, 0.01678, 0.01423, 0.01658, 0.01741, 0.03491, 0.01408, 0.01191, 0.01572, 0.01355, 0.01477, 0.01662, 0.01128, 0.02581, 0.01718, 0.01705, 0.01251, 0.02158, 0.01187, 0.01504, 0.0135, 0.02432, 0.01602, 0.01194, 0.0153, 0.0199, 0.01673, 0.01049, 0.02068, 0.01573, 0.01487, 0.02639, 0.01981, 0.02123, 0.02134, 0.0217, 0.01177},
+			},
+			stakes:                   []float64{1176644.37627, 384623.3607, 394676.13226, 207999.66194, 368582.76542},
+			allListeningCoefficients: []float64{1.0, 1.0, 1.0, 1.0, 1.0},
+			numReputers:              5,
+			want:                     []float64{17.4346, 20.13897, 24.08276, 11.41393, 15.33319},
+			wantErr:                  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := module.GetAllConsensusScores(tt.allLosses, tt.stakes, tt.allListeningCoefficients, tt.numReputers)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetAllConsensusScores() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if !slicesAreApproxEqual(got, tt.want, 1e-2) {
+				t.Errorf("GetAllConsensusScores() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetAllReputersOutput(t *testing.T) {
```

Comment:
I think you should also test the number of iterations as well for diff input, and ensure it iterates the expected number of times. this is an algorithm not commonly if ever found on-chain and for good reason -- it could get quite expensive. so we should be extra sure we're properly bounding the computation here and testing that's the case

Reply:
ticket created!